### PR TITLE
Use `workbench.action.files.openFileFolder` on Mac

### LINKED ISSUE #2846: No dialog pops up after clicking "Open Folder" button
**OS:** Mac
**Build Version:** 20210405.1

**Repro Steps:**

1. Install the docker extension -> F1 -> Type and select "Add Docker Files to Workspace... -> Click "Open Folder" when asking to open a folder or workspace.
2. Check whether open a dialog to select a folder or workspace.

**Expect:**
Pop up a dialog to select a folder or workspace.
Screen Shot 2021-04-06 at 4 08 51 PM

**Actual:**
No dialog pops up.

**More Info:**
This issue doesn't reproduce on Windows/Linux.

### PR DESCRIPTION
**Pull request by bwateratmsft**
Fixes #2846

### CODE REVIEW

File: src/utils/quickPickWorkspaceFolder.ts [hunk 1]
```diff
@@ -21,7 +22,12 @@ export async function quickPickWorkspaceFolder(context: IActionContext, noWorksp
         context.errorHandling.buttons = [
             {
                 callback: async () => {
-                    void vscode.commands.executeCommand('workbench.action.files.openFolder');
+                    if (isMac()) {
```
Comment (karolz-ms) [inline src/utils/quickPickWorkspaceFolder.ts:25]: So VS Code kept open https://github.com/microsoft/vscode/issues/120752 and indicated that `openFolder` might be registered on the Mac in future. Do you think we should revert to doing things uniformly on all platforms once that happens?

**Review by karolz-ms — APPROVED** [review]

File: src/utils/quickPickWorkspaceFolder.ts (hunk 1 above)
Comment (bwateratmsft) [reply src/utils/quickPickWorkspaceFolder.ts:25]: That's a good question; I think the answer depends on whether `openFolder` on Mac forces opening _only_ folders, or if you can still open files. If it does enforce it, I definitely want to switch, in order to consistently enforce that folders must be opened. If it is not enforced--like "Open..."--maybe not.

Comment (karolz-ms) [reply src/utils/quickPickWorkspaceFolder.ts:25]: OK. I will let you decide whether and how you want to track this

Comment (bwateratmsft) [reply src/utils/quickPickWorkspaceFolder.ts:25]: Sounds good. I'm subscribed to the upstream issue ('cause I opened it), so I'll see it if they close it fixed/won't-fix.

**Review by philliphoff — APPROVED** [review]

**Pull request merged**